src/with-translations.js

```diff
@@ -6,45 +6,49 @@
  * @flow
  */
 
-import React, {Component} from 'react';
-import type {ComponentType} from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
+import type {Context} from 'fusion-core';
+
+type TranslateType = (key: string, interpolations?: {[string]: string | number}) => string;
 
 /*
 The `withTranslations` HOC takes an array of translation keys as an argument,
 but does not use it at runtime.
-
 However, these keys are captured by `babel-plugin-i18n` at compile-time by
 Fusion's compiler and the compiler uses generate a map of all translations
 in the app.
-
 The translation map is then exposed by `fusion-plugin-i18n/chunk-translation-map.js`
 */
-export const withTranslations = (/*translationKeys*/) => {
-  return (OriginalComponent: ComponentType<any>) => {
-    class WithTranslations extends Component<any> {
-      translateProp: any;
 
-      constructor(props: any, context: any) {
+export const withTranslations = (
+  translationKeys: string[] /*translationKeys*/
+) => {
+  return <T: {}>(
+    Component: React$ComponentType<T>
+  ): Class<React$Component<$Diff<T, {|translate?: TranslateType|}>, *>> => {
```

Comment:
Ideally we avoid using the [now deprecated existential operator](https://github.com/facebook/flow/releases/tag/v0.72.0) here and below.